var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -343,6 +344,9 @@ def enable_or_disable(variant, keyword='USE', var=None, newer=False):
         enable_or_disable('gloo', newer=True)
         enable_or_disable('tensorpipe')
 
+        if '+debug' in self.spec:
+            env.set('DEBUG', True)
```

Comment:
```suggestion
            env.set('DEBUG', 'ON')
        else:
            env.set('DEBUG', 'OFF')
```
This is a bit more explicit and matches the other env vars we set. It also prevents issues when someone uses `py-torch~debug` but has the `DEBUG` environment variable set in their shell. You should still test this to make sure it behaves the same as other env vars though, I haven't tested it yet.